File: apps/companion-app/scripts/generate-french-tts.js
import fs from 'fs';
import path from 'path';
import { execSync } from 'child_process';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const CURRICULUM_PATH = path.join(__dirname, '../src/data/cScaleCurriculum.js');

const curriculumContent = fs.readFileSync(CURRICULUM_PATH, 'utf-8');

// Use a regex to extract contentFr and audioSnippetFr
const regex = /contentFr:\s*'([^']*)'[\s\S]*?audioSnippetFr:\s*'([^']*)'/g;

let match;
while ((match = regex.exec(curriculumContent)) !== null) {
  const content = match[1];
  const audioFile = match[2]; // e.g. /assets/audio/bertrand_supporting_beams_fr.mp3
  
  // Create output path relative to public dir
  const outPath = path.join(__dirname, '../public', audioFile);
  
  // ensure dir exists
  fs.mkdirSync(path.dirname(outPath), { recursive: true });

  console.log(`Generating: ${audioFile}`);
  
  // use python gtts-cli (assuming it's in the .venv we just created)
  const gttsCmd = `../.venv/bin/gtts-cli "${content.replace(/"/g, '\\"')}" -l fr -o "${outPath}"`;
  
  try {
    execSync(gttsCmd);
    console.log(`Successfully created ${outPath}`);
  } catch (err) {
    console.error(`Error generating ${outPath}:`, err.message);
  }
}

File: apps/companion-app/scripts/generate-french-tts.py
import re
import os
from gtts import gTTS

CURRICULUM_PATH = os.path.join(os.path.dirname(__file__), '../src/data/cScaleCurriculum.js')
PUBLIC_DIR = os.path.join(os.path.dirname(__file__), '../public')

with open(CURRICULUM_PATH, 'r', encoding='utf-8') as f:
    content = f.read()

# Find all contentFr and audioSnippetFr pairs
# pattern: contentFr:\s*'([^']*)'[\s\S]*?audioSnippetFr:\s*'([^']*)'
pattern = re.compile(r"contentFr:\s*'([^']*)'[\s\S]*?audioSnippetFr:\s*'([^']*)'")
matches = pattern.findall(content)

for text, audio_path in matches:
    # Ensure audio path is a relative path to public
    if audio_path.startswith('/'):
        audio_path = audio_path[1:]
    
    out_file = os.path.join(PUBLIC_DIR, audio_path)
    os.makedirs(os.path.dirname(out_file), exist_ok=True)
    
    # Fix escaped quotes in text
    clean_text = text.replace("\\'", "'")
    
    print(f"Generating {out_file}...")
    try:
        tts = gTTS(text=clean_text, lang='fr', slow=False)
        tts.save(out_file)
        print(f"Success: {out_file}")
    except Exception as e:
        print(f"Error generating {out_file}: {e}")

